File: app/src/main/java/website/todds/saundio/windows/library/LibraryViewHolder.java
package website.todds.saundio.windows.library;


import android.database.Cursor;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import website.todds.saundio.R;
import website.todds.saundio.data.SaundioTrack;
import website.todds.saundio.util.DurationUtil;
import website.todds.toddlibs.andrutils.CursorUtil;

import static android.provider.MediaStore.Audio.AudioColumns.DURATION;

public class LibraryViewHolder extends RecyclerView.ViewHolder {

    private View mLayout;
    private TextView mTitle;
    private TextView mArtist;
    private TextView mDuration;

    private SaundioTrack mTrackInfo;

    public LibraryViewHolder(View itemView) {
        super(itemView);

        mLayout = itemView;
        mTitle = itemView.findViewById(R.id.vh_track_title);
        mArtist = itemView.findViewById(R.id.vh_track_artist);
        mDuration = itemView.findViewById(R.id.vh_track_duration);
    }

    public void setData(Cursor cursor) {
        mTrackInfo = new SaundioTrack(cursor);
        mLayout.setTag(mTrackInfo);

        mTitle.setText(mTrackInfo.getTitle());
        mArtist.setText(mTrackInfo.getArtist());
        mDuration.setText(DurationUtil.stampFromMillis(CursorUtil.getLong(cursor, DURATION)));
    }

    public void setOnClickListener(View.OnClickListener clickListener) {
        mLayout.setOnClickListener(clickListener);
    }
}

File: app/src/main/kotlin/website/todds/saundio/windows/library/LibraryViewHolder.kt
package website.todds.saundio.windows.library


import android.database.Cursor
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView

import website.todds.saundio.R
import website.todds.saundio.data.SaundioTrack
import website.todds.saundio.util.DurationUtil
import website.todds.toddlibs.andrutils.CursorUtil

import android.provider.MediaStore.Audio.AudioColumns.DURATION

class LibraryViewHolder(private val mLayout: View) : RecyclerView.ViewHolder(mLayout) {
    private val mTitle: TextView = mLayout.findViewById(R.id.vh_track_title)
    private val mArtist: TextView = mLayout.findViewById(R.id.vh_track_artist)
    private val mDuration: TextView = mLayout.findViewById(R.id.vh_track_duration)

    private var mTrackInfo: SaundioTrack? = null

    fun setData(cursor: Cursor) {
        mTrackInfo = SaundioTrack(cursor)
        mLayout.tag = mTrackInfo

        mTitle.text = mTrackInfo!!.title
        mArtist.text = mTrackInfo!!.artist
        mDuration.text = DurationUtil.stampFromMillis(CursorUtil.getLong(cursor, DURATION))
    }

    fun setOnClickListener(clickListener: View.OnClickListener) {
        mLayout.setOnClickListener(clickListener)
    }
}
